### PR TITLE
Move actions and types around for generation

### PR DESCRIPTION
**Pull request by kalmanb**
Trial moving stuff around so that we can generate the sagas.js from an apidoc spec.

Not for actual merge - just to see what it would be like

### CODE REVIEW

File: src/generated/todo/getTodosSaga.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
+// This file is generated
+
+import { takeEvery } from 'redux-saga';
```
Comment (d4ncer) [inline src/generated/todo/getTodosSaga.js:3]: Can we assume that we will always use `takeEvery`? Are there any cases where we'd prefer to `takeLatest` instead?

File: src/generated/todo/getTodosSaga.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+import { call, put } from 'redux-saga/effects';
+
+// FIXME - implement proper api
+function api(aPathParam, aQueryParam) {
+  return new Promise((resolve) => {
+    // Use superagent
```
Comment (d4ncer) [inline src/generated/todo/getTodosSaga.js:9]: I'd use `superagent` patched with a promise library to get proper Promise APIs. 

Sample patch using `superagent-promise-plugin` and `es6-promise`

``` js
import Promise from 'es6-promise';
import superagentPromisePlugin from 'superagent-promise-plugin';
import request from 'superagent';

Promise.polyfill();
export default superagentPromisePlugin.patch(request);
```

The patched version can be consumed anywhere else in the app like so

``` js
import r from './superagent-patch';

export const fetchTodos = () => r.get('/todos').then(r => r.body.data).catch(e => e);
```

File: src/generated/todo/getTodosSaga.js [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    type: actionTypes.getTodos_failure,
+    payload: err,
+    error: true,
+  }),
+};
+
+function* saga(action) {
+  const { aPathParam, aQueryParam } = action.payload;
+  try {
+    yield put({ type: actions.getTodos_doing });
```
Comment (d4ncer) [inline src/generated/todo/getTodosSaga.js]: Is there any reason this isn't

``` js
yield put(actions.getTodos_doing());
```

File: src/generated/todo/index.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  postTodoSaga.actionTypes
+);
+
+const actions = Object.assign(
+  {},
+  getTodosSaga.actions,
+  postTodoSaga.actions
+);
+
+const sagas = [
```
Comment (d4ncer) [inline src/generated/todo/index.js]: Minor stylistic choice, but I'd prefer

``` js
const sagas = [].concat(
  getTodosSaga.watchingSaga,
  postTodoSaga.watchingSaga,
);
```

File: src/todos/components/MainSection.js [hunk 5]
```diff
@@ -41,10 +44,9 @@ class MainSection extends Component {
   }
 
   componentDidMount() {
-    this.props.actions.fetch();
+    this.props.actions.getTodos_get('dummyPathParam', { dumuyQueryParam: 'someText' });
```
Comment (d4ncer) [inline src/todos/components/MainSection.js]: Spelling

``` js
... dumuyQueryParam: 'someText' ...
```

File: src/generated/todo/getTodosSaga.js (hunk 2 above)
Comment (kalmanb) [inline src/generated/todo/getTodosSaga.js:9]: superagent 2!
https://github.com/visionmedia/superagent/blob/v2.0.0/docs/index.md#promise-and-generator-support

File: src/generated/todo/getTodosSaga.js (hunk 3 above)
Comment (kalmanb) [inline src/generated/todo/getTodosSaga.js]: Good catch

File: src/generated/todo/getTodosSaga.js (hunk 1 above)
Comment (kalmanb) [inline src/generated/todo/getTodosSaga.js:3]: Yeah, I'm not sure. I think we probably need to use it a bit to see which we prefer.

**Comment by d4ncer** [comment]
Awesome, LGTM

**Comment by kalmanb** [comment]
LGTM

**Pull request merged**